litestar/contrib/mongodb_motor/repository/_async.py

```diff
@@ -0,0 +1,440 @@
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of records returned by query, ignoring pagination.
+        """
+        query = self._build_query_from_filters(*filters)
+        query.update(kwargs)
+
+        with wrap_pymongo_exception():
+            cursor = self.collection.find(query)
+            docs = await self._convert_cursor_to_list(cursor)
+            return docs, len(docs)
+
+    async def _convert_cursor_to_list(self, cursor: Any) -> list[DocumentType]:
+        return await _convert_cursor_to_list_async(cursor)
+
+    async def list(self, *filters: FilterTypes, **kwargs: Any) -> list[DocumentType]:
+        """Get a list of instances, optionally filtered.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            The list of instances, after filtering applied.
+        """
+        query = self._build_query_from_filters(*filters)
+        query.update(kwargs)
+
+        with wrap_pymongo_exception():
+            cursor = self.collection.find(query)
+            return await self._convert_cursor_to_list(cursor)
+
+    def filter_collection_by_kwargs(  # type:ignore[override]
+        self,
+        collection: dict[str, Any],
+        /,
+        **kwargs: Any,
+    ) -> dict[str, Any]:
+        """Filter the collection by kwargs.
+
+        Args:
+            collection: A dictionary representing the initial query.
+            **kwargs: key/value pairs that should be added to the query.
+
+        Returns:
+            An updated query dictionary with additional filters.
+        """
+
+        collection.update(kwargs)
+        return collection
+
+    def _build_query_from_filters(self, *filters: FilterTypes) -> dict:
+        """Build a query dictionary from the provided filters.
+
+        Args:
+            *filters: Filters to apply to the collection.
+
+        Returns:
+            A dictionary representing the query.
+        """
+        query = {}
+        for filter_ in filters:
+            if isinstance(filter_, BeforeAfter):
+                query.update(
+                    self._build_before_after_query(filter_.field_name, before=filter_.before, after=filter_.after)
+                )
+            elif isinstance(filter_, CollectionFilter):
+                query.update(self._build_collection_filter_query(filter_.field_name, values=filter_.values))
+            elif isinstance(filter_, SearchFilter):
+                query.update(
+                    self._build_search_filter_query(
+                        filter_.field_name, value=filter_.value, ignore_case=bool(filter_.ignore_case)
+                    )
+                )
+            else:
+                raise RepositoryError(f"Unsupported filter type: {type(filter_)}")
+        return query
+
+    def _build_before_after_query(
```

Comment:
- All `_build_*query` methods return a dict with a single `field_name` key; they can be simplified by returning just the value.
- I'd just inline them, they're called only once and they're not that complex to be extracted into methods.